test/unit/util/locking/RedisResourceLocker.test.ts

```diff
@@ -0,0 +1,124 @@
+// eslint-disable-next-line import/default
+import redis from 'redis';
+import type { RedisClient } from 'redis';
+import { Lock } from 'redlock';
+import { InternalServerError } from '../../../../src/util/errors/InternalServerError';
+import { RedisResourceLocker } from '../../../../src/util/locking/RedisResourceLocker';
+
+// eslint-disable-next-line import/no-named-as-default-member
+redis.createClient = jest.fn().mockReturnValue({
+  on: jest.fn(),
+  quit: jest.fn(),
+  get: jest.fn(),
+});
+
+describe('A RedisResourceLocker', (): void => {
+  let locker: any;
+  const identifier = { path: 'http://test.com/foo' };
+  beforeEach(async(): Promise<void> => {
+    locker = new RedisResourceLocker([ '6379' ]);
+    locker.redlock.lock = jest.fn().mockImplementation(
+      async(resource: string, ttl: number): Promise<Lock> =>
+        new Promise<Lock>((resolve): void => {
+          const lock = new Lock(locker.redlock, resource, 'test', ttl, 100);
+          resolve(lock);
+        }),
+    );
+    locker.redlock.unlock = jest.fn().mockImplementation(
+      async(): Promise<void> =>
+        new Promise<void>((resolve): void => {
+          resolve();
+        }),
+    );
+    locker.redlock.extend = jest.fn().mockImplementation(
+      async(lock: Lock, ttl: number): Promise<Lock> =>
+        new Promise<Lock>((resolve): void => {
+          lock.expiration += ttl;
+          resolve(lock);
+        }),
+    );
+  });
+
+  afterEach((): any => jest.resetAllMocks());
+
+  it('can lock and unlock a resource.', async(): Promise<void> => {
```

Comment:
For this and all tests below: would add some `expect(...).toHaveBeenCalledTimes` and `expect(...).toHaveBeenLastCalledWith` checks to see that the correct redis functions have been called.